### DISCUSSION TITLE
Is there a design principle that prevents exposing channel IDs at the API level?

### DISCUSSION THREAD

**Discussion by HakShak** [rabbitmq/amqp091-go#283] [Q&A] [answered]
We've been running into 
> PRECONDITION_FAILED - delivery acknowledgement on channel NNNN timed out. Timeout value used: 1800000 ms. This timeout value can be configured, see consumers doc guide to learn more

I am making an assumption that the channels IDs are monotonic and the SDK keeps track of them to be in sync with the server. 
But we don't have a way to correlate that channel number when the error is received via a channel close. Is there a way to access it?

**Comment by Zerpet** [comment, marked as answer]
Channel IDs are just an incrementing number, and channels are associated to a connection. In other words, you can have many channels with ID 0, each of them belonging to a different connection. IIRC, the channel ID in the client (SDK) does not match the channel name in the server. In the server, the channel name is something like `<rabbit@some-hostname.3.815.0>`.

To answer your question:  Is there a way to access it? No, not really. Since those IDs are not in sync with the channel name in the server, I don't think it will be very helpful to expose that information.

I understand that it may be difficult to correlate a channel error back to the actual channel just from the [amqp.Error](https://pkg.go.dev/github.com/rabbitmq/amqp091-go#Error) struct. I would recommend that you store all the relevant information in the routine that is going to listen to notifications from [Channel.NotifyClose](https://pkg.go.dev/github.com/rabbitmq/amqp091-go#Channel.NotifyClose), and log that information in your application, if the close notification is ever received.

Hope this helps, fellow capsuleer. Fly dangerous o7

**Reply by HakShak** [reply]
Yeah, that makes sense. Just a bit awkward when dealing with the 30 minute timeout where you can't correlate that with other information that might have been relevant. Time to dump more attributes into tracing!

Also, thanks for [this](https://github.com/rabbitmq/amqp091-go/pull/181), it's already being put to use in New Eden 🚀

**Reply by Zerpet** [reply]
It made my day to know that I've been able to give something back to New Eden, after all the good memories and friends that New Eden gave to me 🙂
